src/web/app/pages-instructor/instructor-home-page/instructor-home-page.component.spec.ts

```diff
@@ -28,4 +70,145 @@ describe('InstructorHomePageComponent', () => {
+    const instructorName: string = '';
+    const courseTabModels: any = {
+      instructorPrivilege,
+      course: defaultCourse,
+      sessionsTableRowModels: [],
+      sessionsTableRowModelsSortBy: SortBy.NONE,
+      sessionsTableRowModelsSortOrder: SortOrder.ASC,
+      hasPopulated: false,
+      isAjaxSuccess: true,
+      isTabExpanded: true,
+    };
+    component.user = instructorName;
+    component.courseTabModels = [courseTabModels];
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with one course with one feedback session with instructor privilege', () => {
+    const instructorName: string = '';
+    const sessionsTableRowModel: any = {
+      feedbackSession,
+      instructorPrivilege,
+      responseRate: '0 / 6',
+      isLoadingResponseRate: false,
+    };
+    const courseTabModels: any = {
+      instructorPrivilege,
+      course: defaultCourse,
+      sessionsTableRowModels: [sessionsTableRowModel],
+      sessionsTableRowModelsSortBy: SortBy.NONE,
+      sessionsTableRowModelsSortOrder: SortOrder.ASC,
+      hasPopulated: true,
+      isAjaxSuccess: true,
+      isTabExpanded: true,
+    };
+    component.user = instructorName;
+    component.courseTabModels = [courseTabModels];
+    fixture.detectChanges();
+    expect(fixture).toMatchSnapshot();
+  });
+
+  it('should snap with one course with two feedback sessions with tutor privilege', () => {
+    const instructorName: string = '';
+    const feedbackSession1: any = {
+      courseId: 'CS3281',
+      timeZone: 'Asia/Singapore',
+      feedbackSessionName: 'Feedback 1',
+      instructions: 'Answer all questions',
+      submissionStartTimestamp: 0,
+      submissionEndTimestamp: 1,
+      gracePeriod: 0,
+      sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
+      customSessionVisibleTimestamp: 0,
+      responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
+      customResponseVisibleTimestamp: 0,
+      submissionStatus: FeedbackSessionSubmissionStatus.CLOSED,
+      publishStatus: FeedbackSessionPublishStatus.PUBLISHED,
+      isClosingEmailEnabled: true,
+      isPublishedEmailEnabled: true,
+      createdAtTimestamp: 0,
+    };
+    const feedbackSession2: any = {
+      courseId: 'CS3281',
+      timeZone: 'Asia/Singapore',
+      feedbackSessionName: 'Feedback 2',
+      instructions: 'Answer all questions',
+      submissionStartTimestamp: 10000,
+      submissionEndTimestamp: 15000,
+      gracePeriod: 100,
+      sessionVisibleSetting: SessionVisibleSetting.AT_OPEN,
+      customSessionVisibleTimestamp: 0,
+      responseVisibleSetting: ResponseVisibleSetting.AT_VISIBLE,
+      customResponseVisibleTimestamp: 0,
+      submissionStatus: FeedbackSessionSubmissionStatus.OPEN,
+      publishStatus: FeedbackSessionPublishStatus.NOT_PUBLISHED,
+      isClosingEmailEnabled: true,
+      isPublishedEmailEnabled: true,
+      createdAtTimestamp: 1000,
+    };
+    const tutorPrivilege: any = {
```

Comment:
I think based on the HTML, it is possible to be able to modify the course, but not the student, and vice versa, and the rendered page is different. I think one of the existing cases should be able to accommodate that situation.

Reply:
Yup, but the changes are independent of one another, so we don't have to permutate all the boolean values for `canModify...`. I believe the current test cases are enough because we already have the snapshots when boolean value is true or false.

Reply:
Gotcha